src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ContainedPartIndex.cs

```diff
@@ -18,17 +18,23 @@ public override void Describe(DescribeContext<ContentItem> context)
                 .When(contentItem => contentItem.Has<ContainedPart>())
                 .Map(contentItem =>
                 {
+                    // Remove index records of soft deleted items.
+                    if (!contentItem.Published && !contentItem.Latest)
```

Comment:
I'm not sure about this one

This means when restoring the content item it is lost from the list, and often lost from the admin controller view, because by default they are not listable.

It should never be lost from the list it is contained in.

Currently I do this restoration as required manually with sql.
This would mean having to then create an index record for the restored item as well.

Now the audit trail module will probably do this automatically (i.e. save it), as the item would still have the `ContainedPart` 

But if there is ever any audit trail, restore content item, from an action in a list, we'd probably need a query on the index (which would now be empty).

So I think no advantage to removing it, but many disadvantages.

Reply:
Yes, i was not sure about contained items, and just because thinking about audit trail

But if audit trail restore a container to re-publish / re-draft it, i thought it would retrieve the ids of the contained item and then restore them knowing their ids, this to re-publish / re-save them

Reply:
Okay, just checked, this is the contained item that knows its parent ;)

So i will revert this for the `containedPartIndex`